### ISSUE TITLE
Separate express middleware from NTLM_Proxy

### ISSUE THREAD

**Issue by mancioshell** [einfallstoll/express-ntlm#41]
Thank you so much for your module. It works great. I'm going to develop an hapi authentication strategy that use ntlm authentication. I would like to reuse your client code interact with NTLM_Proxy but i think it would be better if you could create a separate module for the NTLM_Proxy , and express-ntlm or other node modules could use it for any type of stuff.

Is it possible? In any case, i could give you any help or contributions.

**Comment by einfallstoll** [comment]
I like this idea. Except for the [*real middleware*](https://github.com/einfallstoll/express-ntlm/blob/master/lib/express-ntlm.js#L196) everything could be separated into its own module. This would also be a better solution for #31 without unnecessarily duplicating code.

What's your timeframe on this? I could try to separate the code into a separate module, then we'll test it for your authentication and on a test-branch of express-ntlm and if it's ready we'll ship an update to express-ntlm.

**Comment by mancioshell** [comment]
@einfallstoll I have already separeted NTLM_Proxy wrapping it in a client standalone, and have used it for my hapi authentication strategy module, but i was waiting for your approval to publish it in a pull request on a separeted branch. I have integrated only some parts of NTLM_Proxy in the client (for example i haven't used NO_Proxy), because i need only some functionality, so i need your code review. If it's ok for you, i could publish it in my express-ntlm form on github.
